src/servers/ZoneServer/zoneserver.ts

```diff
@@ -1931,6 +1982,70 @@ export class ZoneServer extends EventEmitter {
   spawnProps(client: Client): void {
     this.spawnNpcCollection(client, this._props);
   }
+  
+  customizeDTO(client: Client): void {
+    const DTOArray: any = [];
+    for (const object in this._destroyable) {
+      const DTO = this._destroyable[object];
+      const DTOinstance = {
+        objectId: DTO.zoneId,
+        unknownString1: "Weapon_Empty.adr",
+      };
+      DTOArray.push(DTOinstance);
+    }
+    for (const object in this._props) {
+      const DTO = this._props[object];
+      const DTOinstance = {
+        objectId: DTO.zoneId,
+        unknownString1: "Weapon_Empty.adr",
+      };
+      DTOArray.push(DTOinstance);
+    }
+    this.sendData(client, "DtoObjectInitialData", {
+      unknownDword1: 1,
+      unknownArray1: DTOArray,
+      unknownArray2: [{}],
+    });
+  }
+  
+  spawnDTOs(client: Client): void {
+    for (const DTO in this._destroyable) {
+      const DTOObject = this._destroyable[DTO];
+      if (
+        isPosInRadius(
+          DTOObject.renderDistance,
+          client.character.state.position,
+          DTOObject.position
+        ) &&
+        !client.spawnedDTOs.includes(DTOObject)
+      ) {
+        client.npcsToSpawn.push(DTOObject);
+        client.spawnedDTOs.push(DTOObject);
+      }
+    }
+  }
+
+  respawnDTOs(): void {
+    for (const DTO in this._destroyableTimestamp) {
+      const DTOObject = this._destroyableTimestamp[DTO];
```

Comment:
beside having a void running on the main thread and checking every worldroutine / serverTicks we should use timeouts for this

Reply:
done